src/Microsoft.AspNet.Mvc.Core/DefaultControllerActionArgumentBinder.cs

```diff
@@ -63,6 +66,21 @@ public class DefaultControllerActionArgumentBinder : IControllerActionArgumentBi
             return actionArguments;
         }
 
+        // Called via reflection.
+        private static void CallPropertyAddRange<TElement>(object target, object source)
+        {
+            var targetCollection = (ICollection<TElement>)target;
+            var sourceCollection = source as IEnumerable<TElement>;
+            if (sourceCollection != null && !targetCollection.IsReadOnly)
+            {
+                targetCollection.Clear();
+                foreach (var item in sourceCollection)
+                {
+                    targetCollection.Add(item);
+                }
+            }
+        }
```

Comment:
One thing that always seems more confusing than necessary when looking through this code is that it just seems excessively generic. Any of the built-in collection types will implement `ICollection` and `IEnumerable` (non-generic), and it's really recommended that user-defined collections do the same.

I'm wondering if we can just simplify our lives by *un-generic-ing* all of the model binding code.

I understand if you want to leave this code as-is for consistency with the legacy, but we should think about simplifying this once we have confidence in our set of tests.